config/config/src/main/java/io/helidon/config/ConfigSourcesRuntime.java

```diff
@@ -37,11 +37,11 @@
 final class ConfigSourcesRuntime {
     private final List<RuntimeWithData> loadedData = new LinkedList<>();
 
-    private List<ConfigSourceRuntimeBase> allSources;
-    private MergingStrategy mergingStrategy;
-    private Consumer<Optional<ObjectNode>> changeListener;
+    private final List<ConfigSourceRuntimeImpl> allSources;
+    private final MergingStrategy mergingStrategy;
+    private volatile Consumer<Optional<ObjectNode>> changeListener;
```

Comment:
Is volatile necessary here? Value is assigned only in synchronized block

Reply:
It is not - the `.onChange() on line 86 is not in synchronized block.

Reply:
missed that